src/precice/impl/ParticipantImpl.cpp

```diff
@@ -1481,6 +1709,18 @@ void ParticipantImpl::computeMappings(std::vector<MappingContext> &contexts, con
       PRECICE_INFO("Computing \"{}\" mapping from mesh \"{}\" to mesh \"{}\" in \"{}\" direction.",
                    context.mapping->getName(), context.mapping->getInputMesh()->getName(), context.mapping->getOutputMesh()->getName(), mappingType);
       context.mapping->computeMapping();
+      anyMappingChanged = true;
+    }
+  }
+  if (anyMappingChanged) {
+    if (mappingType == "write") {
+      for (auto &context : _accessor->writeDataContexts()) {
+        context.initializeMappingDataCache();
+      }
+    } else {
+      for (auto &context : _accessor->readDataContexts()) {
+        context.initializeMappingDataCache();
+      }
```

Comment:
Could be factored into `_accessor`. Then call `_accessor.initializeMappingDataCache(mappingType)` here.